Sandboxed Google Play crashed when attempting to connect to Chrombook

```
type: crash
osVersion: google/oriole/oriole:13/TQ1A.230105.002/2023011000:user/release-keys
package: com.google.android.gms:225014044
process: com.google.android.gms.persistent
processUptime: 32784 + 362 ms
GmsCompatConfig version: 25

java.lang.SecurityException: WifiService: Neither user 10128 nor current process has android.permission.READ_WIFI_CREDENTIAL.
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3021)
	at android.os.Parcel.createException(Parcel.java:3005)
	at android.os.Parcel.readException(Parcel.java:2981)
	at android.os.Parcel.readException(Parcel.java:2923)
	at android.net.wifi.IWifiManager$Stub$Proxy.getPrivilegedConfiguredNetworks(IWifiManager.java:2210)
	at android.net.wifi.WifiManager.getPrivilegedConfiguredNetworks(WifiManager.java:1697)
	at m.byb.c(:com.google.android.gms.optional_better_together@225014057@22.50.14 (100000-0):2)
	at m.byb.h(:com.google.android.gms.optional_better_together@225014057@22.50.14 (100000-0):1)
	at m.byb.run(:com.google.android.gms.optional_better_together@225014057@22.50.14 (100000-0):15)
	at m.aum.c(:com.google.android.gms.optional_better_together@225014057@22.50.14 (100000-0):6)
	at m.aum.run(:com.google.android.gms.optional_better_together@225014057@22.50.14 (100000-0):7)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at aaqv.run(:com.google.android.gms@225014044@22.50.14 (190400-499278674):0)
	at java.lang.Thread.run(Thread.java:1012)
	Suppressed: m.dey: 
Caused by: android.os.RemoteException: Remote stack trace:
	at android.app.ContextImpl.enforce(ContextImpl.java:2285)
	at android.app.ContextImpl.enforceCallingOrSelfPermission(ContextImpl.java:2313)
	at android.content.ContextWrapper.enforceCallingOrSelfPermission(ContextWrapper.java:948)
	at com.android.server.wifi.WifiServiceImpl.enforceReadCredentialPermission(WifiServiceImpl.java:1070)
	at com.android.server.wifi.WifiServiceImpl.getPrivilegedConfiguredNetworks(WifiServiceImpl.java:3029)


```

--- a/core/java/com/android/internal/gmscompat/StubDef.java
+++ b/core/java/com/android/internal/gmscompat/StubDef.java
@@ -79,8 +79,7 @@ public class StubDef implements Parcelable {
         Class stubProxyClass = null;
         String stubProxyMethodName = null;
 
-        // To find out which API call caused the exception, iterate through the stack trace until
-        // the first app's class (app's classes are loaded with PathClassLoader)
+        // Iterate through the stack trace to find out which API call caused the exception
         for (int i = firstIndex; i < stackTrace.length; ++i) {
             StackTraceElement ste = stackTrace[i];
             String className = ste.getClassName();
@@ -88,29 +87,30 @@ public class StubDef implements Parcelable {
             try {
                 class_ = Class.forName(className, false, defaultClassLoader);
             } catch (ClassNotFoundException cnfe) {
-                return null;
+                class_ = null;
             }
 
-            ClassLoader classLoader = class_.getClassLoader();
-            if (classLoader == null) {
-                return null;
-            }
-
-            String loaderName = classLoader.getClass().getName();
-
-            if ("java.lang.BootClassLoader".equals(loaderName)) {
-                if (stubProxyClass == null && className.endsWith("$Stub$Proxy")) {
-                    stubProxyClass = class_;
-                    stubProxyMethodName = ste.getMethodName();
+            if (class_ != null) {
+                ClassLoader classLoader = class_.getClassLoader();
+                if (classLoader == null) {
+                    return null;
                 }
-                continue;
+
+                String loaderName = classLoader.getClass().getName();
+
+                if ("java.lang.BootClassLoader".equals(loaderName)) {
+                    if (stubProxyClass == null && className.endsWith("$Stub$Proxy")) {
+                        stubProxyClass = class_;
+                        stubProxyMethodName = ste.getMethodName();
+                    }
+                    if (!className.startsWith("java.lang.reflect.")) {
+                        // app classes are never loaded with BootClassLoader
+                        continue;
+                    } // else target method is the previous entry that was invoked via reflection
+                }
             }
 
             if (mode == FIND_MODE_Parcel && stubProxyClass == null) {
-                return null;
-            }
-
-            if (!"dalvik.system.PathClassLoader".equals(loaderName)) {
                 return null;
             }
 

